### ISSUE TITLE
Create the data layer

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#26]
### Who

- [ ] Anonymous visitor
- [ ] Patient
- [ ] Doctor
- [ ] Content editor
- [ ] Marketer
- [ ] Administrator
- [x] Developer

### Needs what

To have repositories that shield the client code from Xperience persistence APIs.

### Why

Testability, code maintenance, extensibility.

### Issue expands upon

This issue expands upon another issue(s): #1 

### Blockers

This issue is blocked by the following issue(s): #`issue number`
